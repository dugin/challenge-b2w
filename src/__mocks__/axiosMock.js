import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {planetMock} from "./responseMock";

export const mock = new MockAdapter(axios);

export const URL = 'https://swapi.co/api/';

mock.onGet(`${URL}planets/`).reply(200, {amount: 50, planet: planetMock, isDone: true});

mock.onGet(`${URL}planets/1/`).reply(200, {data: planetMock});


