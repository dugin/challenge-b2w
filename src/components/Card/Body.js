import React from 'react';
import styled from 'styled-components';
import colors from "../../styles/constants/colors";


const Wrapper = styled.section`  
      padding-bottom: 10px;
      border-bottom: 1px solid ${colors.pattensBlue};
      & > p{
        color: ${colors.baliHai};
        font-size: 14px;
        text-align: start;
      }
`;

const constants = ['POPULATION', 'CLIMATE', 'TERRAIN'];

const Body = (props) => {

    return (
        <Wrapper>
            {constants.map(c=> <p key={c}> {c}: </p>)}
        </Wrapper>
    )
};
export default Body;