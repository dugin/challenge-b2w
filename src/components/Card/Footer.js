import React from 'react';
import styled from 'styled-components';
import colors from "../../styles/constants/colors";
import PropTypes from "prop-types";


const Wrapper = styled.section`  
      display: flex;
      justify-content: center;
      & > p{
        color: ${colors.baliHai};
        font-size: 14px;
        text-align: start;
      }
`;

export const Content = styled.span`  
      color: ${colors.sapphire};
      font-size: 16px;
`;


const Footer = (props) => {
    const {appearedInMovie} = props;

    return (
        <Wrapper>
            <p>
                FEATURED IN
                <Content> {appearedInMovie === 0 ? 'NO' : appearedInMovie}</Content>
                {appearedInMovie === 1 ? ' FILM' : ' FILMS'}
            </p>

        </Wrapper>
    )
};

Footer.propTypes = {
    appearedInMovie: PropTypes.number.isRequired,
};
export default Footer;