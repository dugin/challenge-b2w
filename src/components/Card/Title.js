import React from 'react';
import styled from 'styled-components';
import colors from "../../styles/constants/colors";
import PropTypes from "prop-types";


const Wrapper = styled.h1`  
      color: ${colors.brightTurquoise};
      font-weight: 300;
      padding-bottom: 10px;
      border-bottom: 1px solid ${colors.pattensBlue};
`;

const Title = (props) => {

    return (
        <Wrapper>
            {props.text}
        </Wrapper>
    )
};

Title.propTypes = {
    text: PropTypes.string.isRequired,
};
export default Title;