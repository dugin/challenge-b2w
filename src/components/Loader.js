import React from 'react';
import colors from "../styles/constants/colors";
import ReactLoading from 'react-loading';
import styled from 'styled-components';


const Container = styled.section`
 position:absolute;
 z-index: 1;
 width: 100%;
 height: 100%;
 display: flex;
 align-items: center;
 flex-direction: column;
 justify-content: center;
 
 & > *{
 margin-bottom: 10px;
 font-size: 14px;
 color:${colors.sapphire};
 font-weight: 300;
 }
`;



const Loader = (props) => {

    return (
        <Container>
            <ReactLoading type="spin" color={colors.sapphire} delay={0}  height={30} width={30}/>
            <span>Loading...</span>
        </Container>
    )
};
export default Loader;