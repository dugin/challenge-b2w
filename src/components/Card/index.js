import React from 'react';
import styled from 'styled-components';
import Title from "./Title";
import Body from "./Body";
import Footer from "./Footer";
import {Box} from 'grid-styled'


const Wrapper = styled.section`
      background-color: white;
      display:flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      box-shadow: 5px 5px 25px 0 rgba(46,61,73,.2);
      border-radius: 5px;
      position:relative;
`;

const Card = (props) => {

    return (
        <Wrapper>
            <Box width={[
                4 / 5,
            ]}>
                <Title text='PLANET NAME'/>
                <Body/>
                <Footer/>
            </Box>
        </Wrapper>
    )
};
export default Card;