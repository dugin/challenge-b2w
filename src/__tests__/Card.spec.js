import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Card, {Wrapper} from "../components/Card";
import {planetMock} from "../__mocks__/responseMock";
import Loader from "../components/Loader";
import Body from "../components/Card/Body";

describe('Card Component', () => {

    const card = shallow(<Card isLoading={true} planet={planetMock}/>);

    it('renders properly', () => {
        expect(toJson(card)).toMatchSnapshot();
    });

    it('renders the Loader Component', () => {

        expect(card.find(<Loader/>)).toBeTruthy();
    });

    it('renders the "Not Found" error message', () => {

        const card = shallow(<Card isError={true} planet={planetMock}/>);

        expect(card.find('h1').childAt(0).text()).toEqual('Not found');
    });

    it('renders the card with the info', () => {

        const card = shallow(<Card isLoading={false} planet={planetMock}/>);

        expect(card.find(<Body planet={planetMock}/>)).toBeTruthy();
    });

});