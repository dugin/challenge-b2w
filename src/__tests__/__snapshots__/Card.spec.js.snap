// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders properly 1`] = `
<styled.section>
   
  <Loader />
   
</styled.section>
`;
